hazelcast/src/main/java/com/hazelcast/internal/networking/AbstractChannel.java

```diff
@@ -50,13 +50,17 @@
     private final Set<ChannelCloseListener> closeListeners
             = newSetFromMap(new ConcurrentHashMap<ChannelCloseListener, Boolean>());
     private final boolean clientMode;
+    private final SocketAddress remoteAddress;
+    private final SocketAddress localAddress;
 
     @SuppressWarnings("FieldCanBeLocal")
     private volatile int closed = FALSE;
 
     public AbstractChannel(SocketChannel socketChannel, boolean clientMode) {
         this.socketChannel = socketChannel;
         this.clientMode = clientMode;
+        this.remoteAddress = socketChannel.socket().getRemoteSocketAddress();
```

Comment:
Is it guaranteed that remote socket connection established at this point? Is `AbstractChannel ` always created after a successful connection?

Reply:
Yes. The socket channel is established before the channel is registered.